First cut at nucleo_f429zi support

chris-hl opened the pull request:
Needs the following PRs to be applied before this will build..

- [microxrcedds-client](https://github.com/yashi/microxrcedds-client/pull/1)
- [rmw_microxrcedds](https://github.com/yashi/rmw_microxrcedds/pull/1)
- [rcutils](https://github.com/yashi/rcutils/pull/1)

yashi commented on src/main.c:
```diff
@@ -16,14 +17,17 @@
-#define RCCHECK(fn) { rcl_ret_t temp_rc = fn; if((temp_rc != RCL_RET_OK)){printf("Failed status on line %d: %d. Aborting.\n",__LINE__,(int)temp_rc);for(;;){};}}
+#include "udp_transport.h"
+
+#define RCCHECK(fn) { rcl_ret_t temp_rc = fn; if((temp_rc != RCL_RET_OK)){printf("Failed status on line %d: %d. Aborting.\n",__LINE__,(int)temp_rc);for(;;){k_msleep(100);};}}
```
Why do we need 100 ms sleep here? Would you mind to enlighten me?

cj-hl replied:
Ah - for debugging of networking I was using CONFIG_NET_SHELL, which runs in another thread - this spinning at a higher priority stopped that from running.  The sleep yields to other threads

yashi commented on src/udp_transport.c:
```diff
@@ -0,0 +1,98 @@
+#include <uxr/client/transport.h>
```
I'm sure this is a copy of `microros_transports.c` from [micro_ros_zephyr_module](https://github.com/micro-ROS/micro_ros_zephyr_module).  If that's true, please
- create a commit with no contents change.  So that we can diff against original later if needed.
- mention it in the commit message.

yashi commented on src/main.c:
```diff
@@ -3,7 +3,8 @@
 #include <zephyr/kernel.h>
 #include <zephyr/device.h>
 #include <zephyr/devicetree.h>
-#include <zephyr/drivers/gpio.h>
```
Would you mind keeping this?  I know we don't need it but the original main.c has it for unknown reason. And I'd like to keep the main.c as intact as possible.

yashi commented on src/main.c:
```diff
@@ -35,16 +39,20 @@ void timer_callback(rcl_timer_t * timer, int64_t last_call_time)
+#if IS_ENABLED(CONFIG_NET_DHCPV4)
+	struct net_if *iface;
+	iface = net_if_get_default();
+	net_dhcpv4_start(iface);
```
nice :+1:

yashi reviewed (changes requested):
Thanks for the PR.  And I'm excited!

But I have to ask you to split this commit to multiple ones in this PR.

- a commit to import files from [micro_ros_zephyr_module](https://github.com/micro-ROS/micro_ros_zephyr_module).
- a commit to add changes to imported files (if needed).
- a commit to fix `RCCHECK()`
- Add DHCP support.
- a commit to activate building against Nucleo. (.conf and hal)

Oh and have you tested with POSIX arch?  Did it work?  If it does, would you mind mentioning in a commit message?

Hope I'm not asking too much.

chris-hl committed:
Import custom UDP transport from micro_ros_zephyr_module project
See...
URL: https://github.com/micro-ROS/micro_ros_zephyr_module
branch: humble
path: modules/libmicroros/microros_transports/udp/

chris-hl committed:
Rename microros transport files to be more specific

chris-hl committed:
Rename microros transport code to be UDP specific

chris-hl committed:
Add UDP-based microros custom transport

chris-hl committed:
Prevent RCCHECK() from consuming all CPU on error

Allows for other tasks (eg NET_SHELL) to run

chris-hl committed:
Add DHCP support (if configured)

chris-hl committed:
Add config for nucleo_f429zi board

cj-hl commented:
> Thanks for the PR. And I'm excited!
> 
> But I have to ask you to split this commit to multiple ones in this PR.
> 
>     * a commit to import files from [micro_ros_zephyr_module](https://github.com/micro-ROS/micro_ros_zephyr_module).
> 
>     * a commit to add changes to imported files (if needed).
> 
>     * a commit to fix `RCCHECK()`
> 
>     * Add DHCP support.
> 
>     * a commit to activate building against Nucleo. (.conf and hal)
> 
> 
> Oh and have you tested with POSIX arch? Did it work? If it does, would you mind mentioning in a commit message?
> 
> Hope I'm not asking too much.

That should be there now
